src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -227,25 +228,97 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
             EnsureArg.IsNotNull(progress, nameof(progress));
             EnsureArg.IsNotNull(sharedQueryParametersList, nameof(sharedQueryParametersList));
 
-            // Current batch will be used to organize a set of search results into a group so that they can be committed together.
-            string currentBatchId = progress.Page.ToString("d6");
-
             List<Tuple<string, string>> queryParametersList = new List<Tuple<string, string>>(sharedQueryParametersList);
             if (progress.ContinuationToken != null)
             {
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.ContinuationToken, progress.ContinuationToken));
             }
 
-            if (_exportJobRecord.ExportType == ExportJobType.Patient)
+            var requestedResourceTypes = _exportJobRecord.ResourceType?.Split(',');
+            var filteredResources = new HashSet<string>();
+            if (_exportJobRecord.Filters != null)
             {
-                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, KnownResourceTypes.Patient));
+                foreach (var filter in _exportJobRecord.Filters)
+                {
+                    filteredResources.Add(filter.ResourceType.ToUpper(CultureInfo.InvariantCulture));
+                }
             }
-            else if (_exportJobRecord.ExportType == ExportJobType.All && !string.IsNullOrEmpty(_exportJobRecord.ResourceType))
+
+            IAnonymizer anonymizer = IsAnonymizedExportJob() ? await CreateAnonymizerAsync(cancellationToken) : null;
+
+            if (!progress.FilteredSearchesComplete)
             {
-                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, _exportJobRecord.ResourceType));
+                var index = -1;
+                if (_exportJobRecord.Filters != null)
```

Comment:
nit: Can we flip the if condition checks? Since `progress.FilteredSearchesComplete` is `false `by default, we will call `UpdateJobRecordAsync` in order to mark that we have completed filtered searches. If we check for `_exportJobRecord.Filters != null` first, we could avoid that.